07-On demand scan as an API/README.md

```diff
@@ -0,0 +1,312 @@
+5. Select **Add a permission**.
+6. Go to **APIs my organization uses**.
+7. Search your WebAPI client id and select.
+8. Select **Delegated permissions**.
+9. Select permissions.
+10. **Add permissions**.
+
+
+## Step 3: Get administrator consent for WebAPI app registration
+If 'User consent' is restricted to the WebAPI, then WebAPI must have 'Admin consent' granted to expose the APIs.
+Grant admin consent for client app registration:
+1. Go to Azure Portal.
+2. Go to **App Registration**.
+3. Select your WebAPI App Registration.
+4. Go to **API Permissions**.
+5. Select **Add a permission**.
+6. Click **Grant admin consent** for your Tenant.
+
+## Step 4: Generate authentication token to access subscriptions
+User has to generate authentication token in order to use APIs for any subscription.
+There are two ways to generate access tokens:
+- Option 1: Using client credential flow
+- Option 2: Using user authentication code flow
+
+#### Required Az Module
+``` PowerShell
+Install-Module -Name MSAL.PS -AllowClobber -Scope CurrentUser -repository PSGallery
+```
+
+[Back to top…](README.md#On-this-page)
+
+### Using client credential flow
+Client crediential flow uses the client credentials (client id and client secret) to generate the token. Token will be generated against specified SPN (Service Principal Name) and **SPN must have [required](README.md#Required-roles) access over the subscription** to scan or to get the control scan result.
+
+**Steps for 'WebAPI Owner' to grant the permission for requested Client app:**
+1. Go to Azure Portal.
+2. Go to **App Registration**.
+3. Select WebAPI App Registration.
+4. Go to **Add a client application**.
+5. Add client id of request client app.
+6. Enable the check box **Authorized scopes**.
+7. Add application.
+8. Copy scope from 'Scopes'.
+
+
+> In order to generate the token for APIs, you have to get access for the client application from WebAPI owner.
+> 1. Send the client id to WebAPI owner to request access for client application.
+> 2. WebAPI owner will grant the access and share the scope.
+> 3. Use WebAPI scope while generating the access token.
+
+
+Commands to generate the token:
+``` PowerShell
+# Add client secret key of client app registration created in Step-1.
+$ClientSecret = '<client-secret>' | ConvertTo-SecureString -AsPlainText -Force
+
+$token = Get-MsalToken -TenantId '<tenant-id>' -ClientId '<client-id>' -ClientSecret $ClientSecret -Scopes "<WebAPI-scope>/.default"
+
+```
+
+[Back to top…](README.md#On-this-page)
+
+### Using user authentication code flow
+User authentication code flow uses user's crediential to generate the token. User must have access over the subscription  to scan or to get the control scan result.
+
+
+Command to generate the token:
+``` PowerShell
+
+$token = Get-MsalToken -TenantId '<tenant-id>' -ClientId '<client-app-id>' -RedirectUri 'https://localhost' -Scopes '<WebAPI-scope>'
+
+```
+
+[Back to top…](README.md#On-this-page)
+
+### Required roles:
```

Comment:
This can also be added to pre-requisite

Reply:
Added 'required roles' into pre-requisite.